mailpoet/lib/Automation/Integrations/MailPoet/SubjectTransformers/CommentSubjectToSubscriberSubjectTransformer.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\SubjectTransformers;
+
+use MailPoet\Automation\Engine\Data\Subject;
+use MailPoet\Automation\Engine\Integration\SubjectTransformer;
+use MailPoet\Automation\Engine\WordPress;
+use MailPoet\Automation\Integrations\MailPoet\Subjects\SubscriberSubject;
+use MailPoet\Automation\Integrations\WordPress\Subjects\CommentSubject;
+use MailPoet\Subscribers\SubscribersRepository;
+
+class CommentSubjectToSubscriberSubjectTransformer implements SubjectTransformer {
+
+  /** @var WordPress */
+  private $wp;
+
+  /** @var SubscribersRepository */
+  private $subscribersRepository;
+
+  public function __construct(
+    WordPress $wp,
+    SubscribersRepository $subscribersRepository
+  ) {
+    $this->wp = $wp;
+    $this->subscribersRepository = $subscribersRepository;
+  }
+
+  public function transform(Subject $data): ?Subject {
+
+    if ($this->accepts() !== $data->getKey()) {
+      throw new \InvalidArgumentException('Invalid subject type');
+    }
+    $commentId = (int)$data->getArgs()['comment_id'];
+    $comment = $this->wp->getComment($commentId);
+    if (!$comment instanceof \WP_Comment) {
+      return null;
+    }
+    //phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+    $email = $comment->comment_author_email;
+    if (!$this->wp->isEmail($email)) {
+      return null;
+    }
+
+    $subscriber = $this->subscribersRepository->findOneBy(['email' => $email]);
```

Comment:
In WP, can it happen that I would be logged out and comment under your email without logging in? Just wondering if it can be misuses, or if in that case, WP would force me to log in.

Reply:
I just tested, you do not need to be logged in to use e.g. the admin email address. The comment will be stored, depending on the settings even auto-approved, but the user_id will not be associated with the admin :thinking: